.github/workflows/publishVSCode.yml

```diff
@@ -0,0 +1,107 @@
+name: VSCode Publish
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        type: string
+        required: true
+    outputs:
+      result:
+        description: Set to published if the push to the VSCode marketplace was successful
+        value: ${{ jobs.publish.outputs.result }}
+      status: 
+        description: Set to complete if all tasks assoicated with the workflow were successful.
+        value: ${{ jobs.mergeDevelop.outputs.status }}
+
+# There are two outputs that are a result of running this workflow
+# The steps.publsh.outputs.result is an indicator that the pulish to the vscode marketplace succeeded.
+# The steps.mergeDevelop.outputs.status is an indicator that all the publish steps completed.
+jobs: 
+  checkout-code: 
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: '16.13.x'
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        ref: 'main'
+        token: ${{ secrets.IDEE_GH_TOKEN }}
+    - uses: ./.github/actions/gitConfig
+      with:
+        email: ${{ secrets.IDEE_GH_EMAIL }}
+    - run: npm ci
+  updating-sha-for-publish: 
+    name: 'Update SHA256 File'
+    runs-on: ubuntu-latest
+    steps:
+      - run: npm run vscode:sha256
+      - run: scripts/concatenate-sha256.js
+      - run: rm ./SHA256
+      - run: |
+              git add SHA256.md
+              git commit -m "chore: updated SHA256 v${RELEASE_VERSION}"
+      - run: echo "SHA256.md successfully generated and committed."
+  get-artifacts:
+    name: 'Download Artifacts'
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/download-artifact@v3
+        with:
+          name: VS Code Extensions
+          path: ./extensions
+      - name: Display downloaded vsix files
+        run: ls -R ./extensions
+  publish:
+    needs: ['get-artifacts', 'updating-sha-for-publish']
+    outputs: 
+      result: ${{ steps.vsce.outputs.result }}
+    runs-on: ubuntu-latest
+    env: 
+      VSCE_PERSONAL_ACCESS_TOKEN: ${{ secrets.VSCE_PERSONAL_ACCESS_TOKEN }}
+    steps: 
+      - run: find ./extensions -type f -name "*.vsix" -exec vsce publish --pat ${{ env.VSCE_PERSONAL_ACCESS_TOKEN }} --packagePath {} \;
```

Comment:
This is different then the old flow.  The old flow would generate new vsix files then publish those files in each package directory.  Since we are downloading built artifacts from GHA I just make a 1 liner for doing the publish vsce call.